### ISSUE TITLE
Test failures when locale is not set

### ISSUE THREAD

**Issue by zmeggyesi** [eclipse/jetty.project#3001]
As asked in issue #2941 I'm reporting test failures when build is run in an environment without an explicit locale setting:
```
zmeggyesi@build-runner:~$ locale
LANG=
LANGUAGE=
LC_CTYPE="POSIX"
LC_NUMERIC="POSIX"
LC_TIME="POSIX"
LC_COLLATE="POSIX"
LC_MONETARY="POSIX"
LC_MESSAGES="POSIX"
LC_PAPER="POSIX"
LC_NAME="POSIX"
LC_ADDRESS="POSIX"
LC_TELEPHONE="POSIX"
LC_MEASUREMENT="POSIX"
LC_IDENTIFICATION="POSIX"
LC_ALL=
```
Log file of build process: [jetty-install.log](https://github.com/eclipse/jetty.project/files/2494452/jetty-install.log)
As it is somewhat expected, `FileSystemResourceTest.testUtf8Dir` fails when the system locale is not UTF8-capable. With a UTF8 locale set, the build fails at a later point (`NeedWantClientAuthTest.testNeedClientAuthWithoutAuth`) - [jetty-install-locale.log](https://github.com/eclipse/jetty.project/files/2494468/jetty-install-locale.log)

**Comment by joakime** [comment]
Relevant sections from log file (so that others can search for this issue)

```
[ERROR] testUtf8Dir{Class}[1]  Time elapsed: 0.002 s  <<< ERROR!
java.nio.file.InvalidPathException: Malformed input or input contains unmappable characters: b?m
	at org.eclipse.jetty.util.resource.FileSystemResourceTest.testUtf8Dir(FileSystemResourceTest.java:1460)

[ERROR] testUtf8Dir{Class}[2]  Time elapsed: 0.001 s  <<< ERROR!
java.nio.file.InvalidPathException: Malformed input or input contains unmappable characters: b?m
	at org.eclipse.jetty.util.resource.FileSystemResourceTest.testUtf8Dir(FileSystemResourceTest.java:1460)
```

**Comment by zmeggyesi** [comment]
Also relevant message:
```
[ERROR] testNeedClientAuthWithoutAuth  Time elapsed: 5.033 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
	at org.eclipse.jetty.client.ssl.NeedWantClientAuthTest.testNeedClientAuthWithoutAuth(NeedWantClientAuthTest.java:207)
```

**Comment by joakime** [comment]
The `testUtf8Dir` method ...

https://github.com/eclipse/jetty.project/blob/3490add2a96e683c96693fa3c91c3c31c3ca66d1/jetty-util/src/test/java/org/eclipse/jetty/util/resource/FileSystemResourceTest.java#L1455-L1475

I think we are missing a test Assumption on line 1460.
The InvalidPathException should be caught, and a similar `assumeTrue()` to what we have on `testCaret` (but without the `@DisabledOnOs` annotation)

https://github.com/eclipse/jetty.project/blob/3490add2a96e683c96693fa3c91c3c31c3ca66d1/jetty-util/src/test/java/org/eclipse/jetty/util/resource/FileSystemResourceTest.java#L1065-L1088

**Comment by joakime** [comment]
Fixed in commit a10c84c.

The other issue on `NeedWantClientAuthTest.testNeedClientAuthWithoutAuth` is unrelated to the Locale issue.
